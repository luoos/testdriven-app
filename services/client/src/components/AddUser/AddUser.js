import React from 'react';

const AddUser = (props) => {
  return (
    <form onSubmit={(event) => props.submitForm(event)}>
      <div className="field">
        <label className="label is-large">Username</label>
        <div className="control has-icons-left">
          <input
            name="username"
            className="input is-large"
            type="text"
            placeholder="Enter a username"
            required
            value={props.username}
            onChange={props.handleChange}
          />
          <span className="icon is-small is-left">
            <i className="fas fa-user"></i>
          </span>
        </div>
      </div>
      <div className="field">
        <label className="label is-large">Email</label>
        <div className="control has-icons-left">
          <input
            name="email"
            className="input is-large"
            type="email"
            placeholder="Enter an email address"
            required
            value={props.email}
            onChange={props.handleChange}
          />
          <span className="icon is-left">
            <i className="fas fa-envelope"></i>
          </span>
        </div>
      </div>
      <input
        type="submit"
        name="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  );
};

export default AddUser;