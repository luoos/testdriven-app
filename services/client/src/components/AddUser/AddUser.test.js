import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AddUser from './AddUser';

describe('<AddUser />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddUser />);
  });

  it('should contain three inputs', () => {
    const element = wrapper.find('form');
    expect(element.find('input').length).toBe(3);
    expect(element.find('input').get(0).props.name).toBe('username')
    expect(element.find('input').get(1).props.name).toBe('email')
    expect(element.find('input').get(2).props.name).toBe('submit')
  });

  it('AddUser renders a snapshot properly', () => {
    const tree = renderer.create(<AddUser/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})