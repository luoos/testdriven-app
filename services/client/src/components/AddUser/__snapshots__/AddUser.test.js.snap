// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> AddUser renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      className="label is-large"
    >
      Username
    </label>
    <div
      className="control has-icons-left"
    >
      <input
        className="input is-large"
        name="username"
        placeholder="Enter a username"
        required={true}
        type="text"
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fas fa-user"
        />
      </span>
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label is-large"
    >
      Email
    </label>
    <div
      className="control has-icons-left"
    >
      <input
        className="input is-large"
        name="email"
        placeholder="Enter an email address"
        required={true}
        type="email"
      />
      <span
        className="icon is-left"
      >
        <i
          className="fas fa-envelope"
        />
      </span>
    </div>
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    name="submit"
    type="submit"
    value="Submit"
  />
</form>
`;
